src/__swaps__/screens/Swap/hooks/useSwapEstimatedGasLimit.ts

```diff
@@ -98,8 +84,18 @@ export function useSwapEstimatedGasLimit(
       cacheTime: 60 * 1000, // 1min
       notifyOnChangeProps: ['data', 'isFetching'],
       keepPreviousData: true,
-      placeholderData: gasUnits.basic_swap[chainId],
+      enabled: !!chainId && !!quote && !!assetToSell && assetToSell.chainId === chainId,
+      initialData,
+      initialDataUpdatedAt: 0,
       ...config,
     }
   );
+
+  // we keepPreviousData so we can return the previous gasLimit while fetching
+  // which is great when refetching for the same chainId, but we don't want to keep the previous data
+  // when fetching for a different chainId
+  return {
+    data: data && data.chainId === chainId ? data.gasLimit : initialData?.gasLimit,
```

Comment:
leaving a note for the future: when we'd like to upgrade to react-query v5 where we can replace this with using `placeholderData` which gets access to previous values in v5 (currently it does not, which is why we need to do this particular pattern in order to get previous data while fetching if you stay switch assets while on the same chain)




NEW

[8:16](https://rainbowhaus.slack.com/archives/D05JJTKVA0J/p1719443801054779)
which is why I presume the v5 upgrade where you can use placeholderData but also have access to the previous value is the “nicest” way to handle what we want